modules/recording/recording/templates.tsx

```diff
@@ -1,43 +1,71 @@
 import React, { useState, useEffect } from "react";
+import { GlobalCore } from "@/core/module/module.types";
 import templateService, {
   Template,
 } from "@/services/genesis/templates.service";
 import styles from "./styles/templates.module.css";
+import DeleteConfirmation from "@/resources/containers/delete-confirmation";
+import { FaTrash, FaRegPenToSquare } from "react-icons/fa6";
+import MessageHandler from "@/core/message-handler";
+import { useRouter } from "next/router";
+
+const messageHandler = MessageHandler.get();
 
 const Templates = () => {
+  const router = useRouter();
   const [templates, setTemplates] = useState<Template[]>([]);
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [templateId, setTemplateId] = useState<number>();
 
   useEffect(() => {
     fetchTemplates();
-  }, []);
+  }, [router]);
 
   const fetchTemplates = async () => {
     setIsLoading(true);
     try {
       const data = await templateService.getTemplates();
-      setTemplates(data);
+      setTemplates(data as Template[]);
     } catch (error) {
       console.error("Error fetching templates:", error);
     } finally {
       setIsLoading(false);
     }
   };
 
-  const handleDelete = async (id: number) => {
-    if (window.confirm("Are you sure you want to delete this template?")) {
-      try {
-        await templateService.deleteTemplate(id);
-        fetchTemplates();
-      } catch (error) {
-        console.error("Error deleting template:", error);
-      }
+  const handleDelete = (id: number) => {
+    setIsModalOpen(true);
+    setTemplateId(id);
+  };
+
+  const confirmAction = async () => {
+    setIsModalOpen(false);
+
+    const resp = await templateService.deleteTemplate(templateId as number);
+    if (resp) {
+      setTemplates(templates.filter((template) => template.id !== templateId));
+      return messageHandler.handleSuccess("Template deleted");
     }
   };
 
-  const handleAddTemplate = () => {
+  const cancelAction = () => {
+    setIsModalOpen(false);
+  };
+
+  const handleAddTemplate = async () => {
     // Implement add template functionality
-    console.log("Add new template");
+    const data = await templateService.createTemplate({
+      ownerId: "9196981c-e26d-4177-98d1-78492a32e292",
+      name: "Template 1",
```

Comment:
Lets add a empty one, once the save, then we saved it, and we need to validate the name field has value